message_ix/model/MESSAGE/model_solve.gms

```diff
@@ -42,15 +42,20 @@ if (%foresight% = 0,
 * rescale the dual of the emission constraint to account that the constraint is defined on the average year, not total
 EMISSION_CONSTRAINT.m(node,type_emission,type_tec,type_year)$(
         EMISSION_CONSTRAINT.m(node,type_emission,type_tec,type_year) ) =
-    EMISSION_CONSTRAINT.m(node,type_emission,type_tec,type_year) / sum(year$( cat_year(type_year,year) ), 1 ) ;
+    EMISSION_CONSTRAINT.m(node,type_emission,type_tec,type_year)
+        / SUM(year$( cat_year(type_year,year) ), duration_period(year) )
+        * SUM(year$( map_first_period(type_year,year) ), duration_period(year) / df_period(year) * df_year(year) );
```

Comment:
No strong reason, but this seems fishy to me... ` / df_period(year) * df_year(year) `